src/Database/Connection.php

```diff
@@ -280,9 +279,8 @@ public function prepare(Query|string $query): StatementInterface
     {
         return $this->getDisconnectRetry()->run(function () use ($query) {
             $statement = $this->_driver->prepare($query);
-
             if ($this->_logQueries) {
-                $statement = $this->_newLogger($statement);
+                $statement->setLogger($this->getLogger());
```

Comment:
Should we make setLogger() part of `StatementInterface`? Otherwise, we have to type that this is a `Statement` instance.

Reply:
Could we instead come up with a solution which doesn't require passing the logger into the statement? The driver could do the logging after it executes the statement.

Reply:
Added to StatementInterface for now. Will revisit separately.

Reply:
Since statements already have access to the driver, could we get the logger from the driver? If logging is disabled we could use a NullObject to do a no-op.